src/DotNetLightning.Core/Serialize/Features.fs

```diff
@@ -296,6 +298,6 @@ type FeatureBit private (bitArray) =
     interface IComparable with
         member this.CompareTo(o) =
             match o with
-            | :? FeatureBit as fb -> this.CompareTo(fb)
+            | :? FeatureBits as fb -> this.CompareTo fb
             | _ -> -1
```

Comment:
Is it normal to return `-1` when two things can't be compared due to being incompatible types? Or should this be an exception?

Reply:
this is actually how comparisons work in .NET: the CompareTo methods have to return either -1, 1, or 0 (the latter is if they are the same element).

Reply:
I get that the function should normally return -1 / 0 / 1 to indicate less / equal / greater, I'm just wondering if a `FeatureBits` should be considered less than any random `obj` you pass to this function or if it's an error to compare them at all.

Reply:
oh I don't know TBH

Reply:
Hah, indeed, official examples throws `ArgumentException` .  Maybe it is good to fix.

https://docs.microsoft.com/en-us/dotnet/api/system.icomparable?redirectedfrom=MSDN&view=netcore-3.1

Reply:
ok cool I'll add a fix for this while I'm at it